### ISSUE TITLE
XslTransform task not available in .NET Core

### ISSUE THREAD

**Issue by BrunoJuchli** [microsoft/msbuild#3368]
### Steps to reproduce

Project file (test.msbuild)
```xml
<Project>
   <Target Name="Foo">
      <XslTransform />
   </Target>
</Project>
```

Command line
```
dotnet msbuild /t:Foo
```
### Expected  behavior
msbuild complains that not all necessary parameters are given to the task:

>  error MSB4044: The "XslTransform" task was not given a value for the required parameter "...".

(of course, if all required parameter would be given it should execute the task...)

### Actual behavior
>error MSB4036: The "XslTransform" task was not found. Check the following: 1.) The name of the task in the project file is the same as the name of the task class. 2.) The task class is "public" and implements the Microsoft.Build.Framework.ITask interface. 3.) The task is correctly declared with <UsingTask> in the project file, or in the *.tasks files located in the "C:\Program Files\dotnet\sdk\2.1.201" directory.

### Environment data
`dotnet --version` output:
>2.1.201

dotnet msbuild output:
>Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core

Note:
This is similar to https://github.com/Microsoft/msbuild/issues/1731 which was fixed by https://github.com/Microsoft/msbuild/commit/0956cef84e35eb05b2756416059d8cbe00430c5f

**Comment by BrunoJuchli** [comment]
Note, that when I encounter this issue in my actual project configuration, the error output is different:
>error MSB4062: The "Microsoft.Build.Tasks.XslTransformation" task could not be loaded from the assembly Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\work\Erowa.ContractGenerator\source\Erowa.ContractGenerator\Erowa.ContractGenerator.csproj]

So there's actually a `UsingTask` declaration being loaded somewhere. I've been unable to create a minimal repro for this, so far, though.

However, if anyone would like to have a look at the binary log, here it is:
[binlog.zip](https://github.com/Microsoft/msbuild/files/2072321/binlog.zip)

**Comment by rainersigwald** [comment]
This was a task that didn't make the cut way back when we originally ported to .NET Core. My understanding is that .NET Core 2.0 added all the XSL functionality we need (like [`XslCompiledTransform`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xsl.xslcompiledtransform?view=netstandard-2.0)), so this should be easy to bring up for all target frameworks.

**Comment by rainersigwald** [comment]
Unfortunately, it's more complicated than just turning the task on again. I tried in https://github.com/rainersigwald/msbuild/tree/xsltransform but the tests give errors

```
XslTransformation_Tests.cs(1161,100): error CS0117: 'AssemblyBuilderAccess' does not contain a definition for 'Save' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1161,47): error CS1061: 'AppDomain' does not contain a definition for 'DefineDynamicAssembly' and no extension method 'DefineDynamicAssembly' accepting a first argument of type 'AppDomain' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1170,45): error CS1501: No overload for method 'DefineDynamicModule' takes 3 arguments [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1187,51): error CS0117: 'XslCompiledTransform' does not contain a definition for 'CompileToType' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1197,21): error CS1061: 'AssemblyBuilder' does not contain a definition for 'Save' and no extension method 'Save' accepting a first argument of type 'AssemblyBuilder' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1217,100): error CS0117: 'AssemblyBuilderAccess' does not contain a definition for 'Save' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1217,47): error CS1061: 'AppDomain' does not contain a definition for 'DefineDynamicAssembly' and no extension method 'DefineDynamicAssembly' accepting a first argument of type 'AppDomain' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1219,45): error CS1501: No overload for method 'DefineDynamicModule' takes 3 arguments [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1233,21): error CS1061: 'AssemblyBuilder' does not contain a definition for 'Save' and no extension method 'Save' accepting a first argument of type 'AssemblyBuilder' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1161,100): error CS0117: 'AssemblyBuilderAccess' does not contain a definition for 'Save' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1161,47): error CS1061: 'AppDomain' does not contain a definition for 'DefineDynamicAssembly' and no extension method 'DefineDynamicAssembly' accepting a first argument of type 'AppDomain' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1170,45): error CS1501: No overload for method 'DefineDynamicModule' takes 3 arguments [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1187,51): error CS0117: 'XslCompiledTransform' does not contain a definition for 'CompileToType' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1197,21): error CS1061: 'AssemblyBuilder' does not contain a definition for 'Save' and no extension method 'Save' accepting a first argument of type 'AssemblyBuilder' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1217,100): error CS0117: 'AssemblyBuilderAccess' does not contain a definition for 'Save' [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1217,47): error CS1061: 'AppDomain' does not contain a definition for 'DefineDynamicAssembly' and no extension method 'DefineDynamicAssembly' accepting a first argument of type 'AppDomain' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1219,45): error CS1501: No overload for method 'DefineDynamicModule' takes 3 arguments [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
XslTransformation_Tests.cs(1233,21): error CS1061: 'AssemblyBuilder' does not contain a definition for 'Save' and no extension method 'Save' accepting a first argument of type 'AssemblyBuilder' could be found (are you missing a using directive or an assembly reference?) [/Users/raines/src/msbuild/src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj]
```

**Comment by rainersigwald** [comment]
I think I didn't notice that those errors are all in tests. If those test compilation failures are the only problem, we should consider disabling only those tests on Core, or rewriting them so that they work everywhere.

**Comment by sven-n** [comment]
Hello, I still have the same issue on my .net core 3.0 project. Is there a fix in sight yet?

## Environment data

> ```dotnet --version``` output:

    3.0.100

> ```dotnet msbuild``` output:

    Microsoft (R)-Build-Engine, Version 16.3.0+0f4c62fea für .NET Core

**Comment by BenVillalobos** [comment]
@sven-n yes! a fix is in the works and is planned to be released in 16.4
